controllers/configmap_controller.go

```diff
@@ -189,6 +202,8 @@ func (r *ConfigMapReconciler) Reconcile(ctx context.Context,
 		return ctrl.Result{}, r.reconcileNodes(ctx, configMap)
 	case certificates.KubeAPIServerServingCAConfigMapName:
 		return ctrl.Result{}, r.reconcileKubeletClientCA(ctx, configMap)
+	case proxyCertsConfigMap:
+		return ctrl.Result{}, r.reconcileProxyCertsCM(ctx, configMap)
```

Comment:
Nope, in the case of a delete event, lines 190-192 will be executed

Reply:
Okay, it seems that logic should be moved into reconcileProxyCertsCM, as that is code specific to reconciling the ProxyCertsCM

Reply:
Let me know if you feel strongly about this, I can do a small refactor.

For context, I followed the existing pattern to only enter the switch statement if the CM is set. See comment above the switch `// At this point configMap will be set properly`. We do something similar for the services CM (lines 184-189).

Reply:
Right, but I don't think the existing pattern makes sense. At least in this case.

You're introducing a new function to reconcile the CM, but you're creating the CM setting the state as we want it to be beforehand. So there's not going to be a reason to call reconcileProxyCertsCM, in the delete case it will always be a no-op as the state has already been reconciled beforehand.

Seeing that it has been deleted and re-creating it is reconciling.